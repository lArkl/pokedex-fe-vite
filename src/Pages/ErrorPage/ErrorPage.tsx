import { FC } from 'react'
import Typography from '../../components/Typography/Typography'
import styles from './ErrorPage.module.scss'

const ErrorPage: FC = () => {
  return (
    <div className={styles.container} aria-label="error">
      <Typography variant="xl">Something went wrong...</Typography>
      <Typography variant="lg">Try reloading the page in some seconds</Typography>
      <Typography variant="md">If this persists, contact to IT</Typography>
    </div>
  )
}

export default ErrorPage
