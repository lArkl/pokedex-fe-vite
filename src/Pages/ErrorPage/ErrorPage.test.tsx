import { render, screen } from '@testing-library/react'
import ErrorPage from './ErrorPage'

describe('ErrorPage', () => {
  it('renders component', async () => {
    const { asFragment } = render(<ErrorPage />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('shows message', async () => {
    render(<ErrorPage />)

    expect(screen.getByLabelText('error')).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /something went wrong/i })).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /reload/i })).toBeInTheDocument()
    expect(screen.getByText(/contact/)).toBeInTheDocument()
  })
})
