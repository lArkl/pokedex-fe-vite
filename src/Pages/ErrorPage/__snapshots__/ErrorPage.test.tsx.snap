// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage > renders component 1`] = `
<DocumentFragment>
  <div
    aria-label="error"
    class="_container_4d13f2"
  >
    <h1
      class="_xl_bcfef7"
    >
      Something went wrong...
    </h1>
    <h2
      class="_lg_bcfef7"
    >
      Try reloading the page in some seconds
    </h2>
    <p
      class="_md_bcfef7"
    >
      If this persists, contact to IT
    </p>
  </div>
</DocumentFragment>
`;
